client/my-sites/checkout/index.js

```diff
@@ -83,6 +98,15 @@ export default function () {
 	);
 
 	page( '/checkout/jetpack/:siteSlug/:productSlug', checkout, makeLayout, clientRender );
+
+	page(
+		`/checkout/jetpack/:siteSlug/:productSlug/${ getLanguageRouteParam() }`,
```

Comment:
Same here: let's add the locale handling to the original route instead of creating a duplicate one.